plugins/provisioning-terraform-aws-core/src/index.ts

```diff
@@ -0,0 +1,134 @@
+import type {
+  AmplicationPlugin,
+  CreateServerParams,
+  DsgContext,
+  Events,
+  ModuleMap,
+} from "@amplication/code-gen-types";
+import {
+  nameKey,
+  regionIdentifierKey,
+  vpcCidrBlockKey,
+  enableDnsHostnamesKey,
+  enableDnsSupportKey,
+  enableNatGatewayKey,
+  singleNatGatewayKey,
+  environmentKey,
+  backendKey,
+} from "./constants";
+import { join } from "node:path";
+import { getPluginSettings } from "./utils";
+import { EventNames } from "@amplication/code-gen-types";
+import { resolve } from "path";
+import { kebabCase } from "lodash";
+
+class TerraformAwsCorePlugin implements AmplicationPlugin {
+  /**
+   * This is mandatory function that returns an object with the event name. Each event can have before or/and after
+   */
+  register(): Events {
+    return {
+      [EventNames.CreateServer]: {
+        after: this.afterCreateServer,
+      },
+    };
+  }
+
+  async afterCreateServer(
+    context: DsgContext,
+    eventParams: CreateServerParams,
+    modules: ModuleMap
+  ): Promise<ModuleMap> {
+    context.logger.info("Generating Terraform Amazon Web Services Core ...");
+
+    // determine the name of the service which will be used as the name for the workflow
+    // workflow names must be lower case letters and numbers. words may be separated with dashes (-):
+    const serviceName = kebabCase(context.resourceInfo?.name);
+
+    if (!serviceName) {
+      throw new Error("Service name is undefined");
+    }
+
+    // fetch the plugin specific settings and merge them with the defaults
+    const settings = getPluginSettings(context.pluginInstallations);
+
+    /**
+     * save the renderedOutput to the desired directory the options are on the root of the repository
+     * and within the directory of the services itself setting "root_directory":
+     *
+     *    option 1 (value: true):  /<directory_name_value>/<directory_name_value>
+     *    option 2 (value: false): /[optional: mono_prefix]/<service_name>/<directory_name_value>
+     */
+
+    let terraformDirectoryPath: string = "";
+    const rootDirectoryPath: string = "./";
+
+    if (settings.root_level === true) {
+      terraformDirectoryPath = join(rootDirectoryPath, settings.directory_name);
+    } else if (settings.root_level === false) {
+      terraformDirectoryPath = join(
+        context.serverDirectories.baseDirectory,
+        settings.directory_name
+      );
+    } else {
```

Comment:
This part with the additional "else" and explicit testing of "else if ... === false" is not needed since the `getPluginSettings()` function already returned a typed object defined in `types.ts` and it specifies that `root_level` is a boolean.

Reply:
reworked into a `conditional operator`

Reply:
I don't think you committed this change if you made it